cmd/codegen/generator/go/templates/module_funcs.go

```diff
@@ -276,6 +289,26 @@ func (ps *parseState) parseParamSpecVar(field *types.Var, docComment string, lin
 			optional, _ = strconv.ParseBool(v)
 		}
 	}
+	defaultPath := ""
+	if v, ok := pragmas["defaultPath"]; ok {
```

Comment:
I'm not a huge fan of using `defaultPath` as the pragma here (even though it's distinct at the API level). Could we just use `default` here?

It doesn't make sense to use `defaultPath` on a non File/Directory arg, and it doesn't make sense to use `default` on a File/Directory arg: we should condense this down into one `default`.

Reply:
He's following the design in:
- https://github.com/dagger/dagger/issues/7647#issuecomment-2201921231

Best to bikeshed this in the issue.

Reply:
FTR, I agree. It's Python and Go that need a distinction since they do support default values in parameters, but in Go it's simpler if it's the same (i.e., `default`) as it doesn't have the same constraints here. The only downside is it's less clear when comparing or porting code from one language to another (consistency).